test/unit/sdam/monitor.test.ts

```diff
@@ -306,6 +307,47 @@ describe('monitoring', function () {
         }, minHeartbeatFrequencyMS);
       });
     });
+
+    describe('roundTripTime', function () {
+      const table = [
+        {
+          serverMonitoringMode: 'stream',
+          topologyVersion: {
+            processId: new ObjectId(),
+            counter: new Long(0, 0)
+          }
+        },
+        { serverMonitoringMode: 'poll', topologyVersion: undefined }
+      ];
+      for (const { serverMonitoringMode, topologyVersion } of table) {
+        context(`when serverMonitoringMode = ${serverMonitoringMode}`, () => {
+          context('when more than one heartbeatSucceededEvent has been captured', () => {
+            const heartbeatDurationMS = 250;
+            it('correctly returns the mean of the heartbeat durations', async () => {
+              mockServer.setMessageHandler(request => {
+                setTimeout(
+                  () => request.reply(Object.assign({ helloOk: true }, mock.HELLO)),
+                  heartbeatDurationMS
+                );
+              });
+              const server = new MockServer(mockServer.address());
+              if (topologyVersion) server.description.topologyVersion = topologyVersion;
+              monitor = new Monitor(server as any, { serverMonitoringMode } as any);
+              monitor.connect();
+
+              for (let i = 0; i < 5; i++) {
+                await once(monitor, 'serverHeartbeatSucceeded');
+                monitor.requestCheck();
+              }
+              expect(monitor.roundTripTime).to.be.greaterThanOrEqual(heartbeatDurationMS);
+              monitor.close();
+            });
+          });
+        });
+      }
+    });
+
+    //describe('minRoundTripTime');
```

Comment:
Do we want 'minRoundTripTime' coverage?

Reply:
Forgot to remove this, but I think we have enough coverage in for this with the `MovingWindow` class, which I'll rename and possibly move to the `monitor.ts` since that's the only place it's used and it implements spec-required behaviour.